Bottom: c11c3449ead7488edd95fa44ab5af9964c3fc223
Top:    952fe0387a7f513dd65abdee45a9578c774eb4c4
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-03-16 20:02:42 +0800

vhost_net: switch to use __skb_array_len_with_tag()

Signed-off-by: Jason Wang <jasowang@redhat.com>


---

diff --git a/drivers/vhost/net.c b/drivers/vhost/net.c
index c7f3922..b5994a6 100644
--- a/drivers/vhost/net.c
+++ b/drivers/vhost/net.c
@@ -529,7 +529,6 @@ static int peek_head_len_batched(struct vhost_net_virtqueue *rvq)
 static int peek_head_len(struct vhost_net_virtqueue *rvq, struct sock *sk)
 {
 	struct socket *sock = sk->sk_socket;
-	struct sk_buff *head;
 	int len = 0;
 	unsigned long flags;
 
@@ -540,14 +539,7 @@ static int peek_head_len(struct vhost_net_virtqueue *rvq, struct sock *sk)
 		return sock->ops->peek_len(sock);
 
 	spin_lock_irqsave(&sk->sk_receive_queue.lock, flags);
-	/* FIXME: duplicated! */
-	head = skb_peek(&sk->sk_receive_queue);
-	if (likely(head)) {
-		len = head->len;
-		if (skb_vlan_tag_present(head))
-			len += VLAN_HLEN;
-	}
-
+	len = __skb_array_len_with_tag(skb_peek(&sk->sk_receive_queue));
 	spin_unlock_irqrestore(&sk->sk_receive_queue.lock, flags);
 	return len;
 }
